crypto/keys/keybase_base.go

```diff
@@ -124,7 +124,7 @@ func (kb baseKeybase) DecodeSignature(info Info, msg []byte) (sig []byte, pub tm
 		return nil, nil, err
 	}
 
-	if err := CryptoCdc.UnmarshalBinaryBare([]byte(signed), sig); err != nil {
+	if err := CryptoCdc.UnmarshalBinaryLengthPrefixed([]byte(signed), sig); err != nil {
```

Comment:
Would you mind to write a small test case for `DecodeSignature()` please? So that next time we could spot similar problems early

Reply:
Nevermind @jgimeno, it's not easy to test this - see `buf := bufio.NewReader(os.Stdin)`. This may take some re-design of baseKeybase